File: OpenKlaster/CommonService/src/main/java/com/openklaster/common/model/Installation.java
package com.openklaster.common.model;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Installation {
    @JsonAlias({ "installationId", "_id" })
    @JsonProperty("_id")
    private String _id;
    private String username;
    private Inverter inverter;
    private Load load;
    private Source source;
    private InstallationType installationType;
    private Double longitude;
    private Double latitude;
    private String description;

    public Installation(String _id, InstallationType type, Inverter inverter, Load load, Source source) {
        this(_id, type);
        this.source = source;
        this.inverter = inverter;
        this.load = load;
    }

    public Installation(String _id, InstallationType type) {
        this._id = _id;
        this.installationType = type;
    }
}

File: OpenKlasterWeb/src/app/model/Installation.ts
import {Load} from './Load';
import {Inverter} from './Inverter';
import {Source} from './Source';

export class Installation {
  installationId: string;
  installationType: string;
  longitude: number;
  latitude: number;
  description: string;
  load: Load;
  inverter: Inverter;
  source: Source;

  constructor() {
    this.load = new Load();
    this.inverter = new Inverter();
    this.source = new Source();
  }


}
